experimental/web/sample_webgpu/main.c

```diff
@@ -0,0 +1,440 @@
+    }
+  }
+
+  iree_vm_list_resize(variants_list, 0);
+
+  return iree_ok_status();
+}
+
+const char* call_function(iree_program_state_t* program_state,
+                          const char* function_name, const char* inputs,
+                          int iterations) {
+  iree_status_t status = iree_ok_status();
+
+  // Fully qualify the function name. This sample only supports loading one
+  // module (i.e. 'program') per session, so we can do this.
+  iree_string_builder_t name_builder;
+  iree_string_builder_initialize(iree_allocator_system(), &name_builder);
+  if (iree_status_is_ok(status)) {
+    iree_string_view_t module_name = iree_vm_module_name(program_state->module);
+    status = iree_string_builder_append_format(&name_builder, "%.*s.%s",
+                                               (int)module_name.size,
+                                               module_name.data, function_name);
+  }
+
+  iree_runtime_call_t call;
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_call_initialize_by_name(
+        program_state->session, iree_string_builder_view(&name_builder), &call);
+  }
+  iree_string_builder_deinitialize(&name_builder);
+
+  if (iree_status_is_ok(status)) {
+    status = parse_inputs_into_call(
+        &call, iree_runtime_session_device_allocator(program_state->session),
+        iree_make_cstring_view(inputs));
+  }
+
+  // Note: Timing has ~millisecond precision on the web to mitigate timing /
+  // side-channel security threats.
+  // https://developer.mozilla.org/en-US/docs/Web/API/Performance/now#reduced_time_precision
+  iree_time_t start_time = iree_time_now();
+  for (int i = 0; i < iterations; ++i) {
+    if (iree_status_is_ok(status)) {
+      status = iree_runtime_call_invoke(&call, /*flags=*/0);
+    }
+  }
+  iree_time_t end_time = iree_time_now();
+  iree_time_t time_elapsed = end_time - start_time;
+
+  iree_string_builder_t outputs_builder;
+  iree_string_builder_initialize(iree_allocator_system(), &outputs_builder);
+
+  // Output a JSON object as a string:
+  // {
+  //   "total_invoke_time_ms": [number],
+  //   "outputs": [semicolon delimited list of formatted outputs]
+  // }
+  if (iree_status_is_ok(status)) {
+    status = iree_string_builder_append_format(
+        &outputs_builder,
+        "{ \"total_invoke_time_ms\": %" PRId64 ", \"outputs\": \"",
+        time_elapsed / 1000000);
+  }
+  if (iree_status_is_ok(status)) {
+    status = print_outputs_from_call(&call, &outputs_builder);
+  }
+  if (iree_status_is_ok(status)) {
+    status = iree_string_builder_append_cstring(&outputs_builder, "\"}");
+  }
+
+  if (!iree_status_is_ok(status)) {
+    iree_string_builder_deinitialize(&outputs_builder);
+    iree_status_fprint(stderr, status);
+    iree_status_free(status);
+    return "";
+  }
+
+  // Note: this leaks the buffer. It's up to the caller to free it after use.
+  return iree_string_builder_buffer(&outputs_builder);
```

Comment:
this is really shady - I know it's a sample but I think it's worth fixing this here before it proliferates/we forget :)

if you returned `strdup(iree_string_builder_buffer(...))` and then deinitialized the builder the caller could then call _free to free it when done

Reply:
Thanks, I fixed that leak and added a link with some context.